pyincore_viz/geoutil.py

```diff
@@ -1094,6 +1094,78 @@ def get_ipyleaflet_heatmap(locations=None, radius=10, blur=10, max=1, name=""):
 
         return heatmap
 
+    @staticmethod
+    def plot_mulitple_vector_dataset(dataset_list, zoom_level=10):
+        geodata_dic_list = []
+        title_list = []
+        bbox = None
+        # check if the dataset is geodataset and convert dataset to geodataframe
+        for dataset in dataset_list:
+            try:
+                tmp_gpd = gpd.read_file(dataset.local_file_path)
+                tmp_min_x = tmp_gpd.bounds.minx.mean()
+                tmp_min_y = tmp_gpd.bounds.miny.mean()
+                tmp_max_x = tmp_gpd.bounds.maxx.mean()
+                tmp_max_y = tmp_gpd.bounds.maxy.mean()
+
+                if bbox is None:
+                    bbox = [tmp_min_x, tmp_min_y, tmp_max_x, tmp_max_y]
+                tmp_bbox = [tmp_min_x, tmp_min_y, tmp_max_x, tmp_max_y]
+
+                if bbox[0] >= tmp_bbox[0]:
+                    bbox[0] = tmp_bbox[0]
+                if bbox[1] >= tmp_bbox[1]:
+                    bbox[1] = tmp_bbox[1]
+                if bbox[2] <= tmp_bbox[2]:
+                    bbox[2] = tmp_bbox[2]
+                if bbox[3] <= tmp_bbox[3]:
+                    bbox[3] = tmp_bbox[3]
+
+                # skim geodataframe only for needed fields
+                tmp_fld_list = ['geometry']
+                tmp_gpd_skimmed = tmp_gpd[tmp_fld_list]
+                tmp_geo_data_dic = json.loads(tmp_gpd_skimmed.to_json())
+                geodata_dic_list.append(tmp_geo_data_dic)
+                title_list.append(dataset.metadata["title"])
+
+            except Exception:
+                raise("The dataset might not be a vector geodataset or field name might not be correct.")
+
+        # calculate center point
+        center_x = ((bbox[2] - bbox[0]) / 2) + bbox[0]
+        center_y = ((bbox[3] - bbox[1]) / 2) + bbox[1]
+
+        out_map = ipylft.Map(center=(center_y, center_x), zoom=zoom_level,
+                             crs=projections.EPSG3857, scroll_wheel_zoom=True)
+
+        for geodata_dic, title in zip(geodata_dic_list, title_list):
+            # add data to  map
+            tmp_layer = ipylft.GeoJSON(
+                data=geodata_dic,
+                style={
+                    'opacity': 1, 'fillOpacity': 0.8, 'weight': 1
+                },
+                hover_style={
+                    'color': 'white', 'dashArray': '0', 'fillOpacity': 0.5
+                },
+                style_callback=GeoUtil.random_color,
+                name=title
+            )
+
+            out_map.add_layer(tmp_layer)
+
+        out_map.add_control(ipylft.LayersControl(position='topright'))
+        out_map.add_control(ipylft.FullScreenControl(position='topright'))
+
+        return out_map
+
+    @staticmethod
+    def random_color(feature):
+        return {
```

Comment:
`feature` is not used here?

Reply:
It has to be in there. Otherwise, it is making an error since this is call back for each feature

Reply:
oh gotcha